File: Java-Core/src/main/java/presto/python/PythonNativeCall.java
package presto.python;

import presto.error.PrestoError;
import presto.error.SyntaxError;
import presto.runtime.Context;
import presto.statement.NativeCall;
import presto.type.IType;
import presto.type.VoidType;
import presto.utils.CodeWriter;
import presto.value.IValue;

public abstract class PythonNativeCall extends NativeCall {

	PythonStatement statement;
	
	public PythonNativeCall(PythonStatement statement) {
		this.statement = statement;
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		statement.toDialect(writer);
	}

	@Override
	public IType check(Context context) throws SyntaxError {
		return VoidType.instance(); // TODO
	}
	
	@Override
	public IValue interpret(Context context) throws PrestoError {
		throw new RuntimeException("Should never get there!");
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/python/PythonNativeCall.js
var NativeCall = require("../statement/NativeCall").NativeCall;

function PythonNativeCall(statement) {
	NativeCall.call(this);
	this.statement = statement;
	return this;
}

PythonNativeCall.prototype = Object.create(NativeCall.prototype);
PythonNativeCall.prototype.constructor = PythonNativeCall;

PythonNativeCall.prototype.toDialect = function(writer) {
    this.statement.toDialect(writer);
}

exports.PythonNativeCall = PythonNativeCall;
